[ONNX] Disable ONNX ceil_mode and count_include_pad to align torch ceil_mode results in corner case

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #87893
* __->__ #87892

ONNX and PyTorch has different equation on pooling and different strategy on ceil_mode, which leads to discrepancy on corner case (#71549 ). 
Specifically, PyTorch avereage pooling is not following [the equation on documentation](https://pytorch.org/docs/stable/generated/torch.nn.AvgPool2d.html), it allows sliding window to go off-bound instead, if they start within the left padding or the input (in NOTE section). More details can be found in #57178.

This PR changes avgpool in opset 10 and 11 back the way as opset 9, which it stops using ceil_mode and count_include_pad  in onnx::AveragePool

A comprehensive test for all combinations of parameters can be found in the next PR. #87893

titaiwangms committed:
[ONNX] Disable ONNX ceil_mode and count_include_pad to aligntorch ceil_mode results in corner case

[ghstack-poisoned]

BowenBao reviewed (approved):
Thanks for digging into this! You'd need to update expect file to pass CI.

titaiwangms committed:
Update on "[ONNX] Disable ONNX ceil_mode and count_include_pad to aligntorch ceil_mode results in corner case"


ONNX and PyTorch has different equation on pooling and different strategy on ceil_mode, which leads to discrepancy on corner case (#71549 ). 
Specifically, PyTorch avereage pooling is not following [the equation on documentation](https://pytorch.org/docs/stable/generated/torch.nn.AvgPool2d.html), it allows sliding window to go off-bound instead, if they start within the left padding or the input (in NOTE section). More details can be found in #57178.

This PR changes avgpool in opset 10 and 11 back the way as opset 9, which it stops using ceil_mode and count_include_pad  in onnx::AveragePool

A comprehensive test for all combinations of parameters can be found in the next PR. #87893 

[ghstack-poisoned]

AllenTiTaiWang commented:
@pytorchbot merge -g

AllenTiTaiWang commented:
@pytorchbot meege -g

AllenTiTaiWang commented:
@pytorchbot merge -g
